### PR TITLE
[msbuild] Added a PropertyListEditor task which works like PlistBuddy

### PR DESCRIPTION
**Pull request by jstedfast**
This is a convenience Task for customers and isn't currently used
by the core MSBuild targets.

### CODE REVIEW

File: msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs [hunk 1]
```diff
@@ -391,7 +391,7 @@ protected override string GenerateCommandLineCommands ()
 				args.Add ("--extension");
 
 			if (Debug) {
-				if (FastDev && IPhoneSdks.MonoTouch.SupportsFastDev)
+				if (FastDev)
```
Comment (spouliot) [inline msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs]: this change seems unrelated to commit

File: msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs [hunk 2]
```diff
@@ -427,10 +427,8 @@ protected override string GenerateCommandLineCommands ()
 			if (UseFloat32 /* We want to compile 32-bit floating point code to use 32-bit floating point operations */)
 				args.Add ("--aot-options=-O=float32");
 
-			if (IPhoneSdks.MonoTouch.SupportsGenericValueTypeSharing) {
-				if (!EnableGenericValueTypeSharing)
-					args.Add ("--gsharedvt=false");
-			}
+			if (!EnableGenericValueTypeSharing)
+				args.Add ("--gsharedvt=false");
```
Comment (spouliot) [inline msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs]: this change seems unrelated to commit

File: msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs (hunk 1 above)
Comment (jstedfast) [inline msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs]: It's required due to the bump of Xamarin.MacDev which removed the SupportsX properties (as they aren't needed, all of them are for versions older than 10.0).

File: msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs (hunk 2 above)
Comment (jstedfast) [inline msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs]: Same as above.

File: msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs (hunk 1 above)
Comment (spouliot) [inline msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs]: ok, but ideally this should have been a different PR, i.e. one with bump and API adjustment and the 2nd for the new task

Comment (jstedfast) [inline msbuild/Xamarin.iOS.Tasks.Core/Tasks/MTouchTaskBase.cs]: True, I probably should have split them.

**Comment by VincentDondain** [comment]
LGTM 👍 nice changes.
